### ISSUE TITLE
Pre-check: destination is not compressed

### ISSUE THREAD

**Issue by Kage-Yami** [pxlrbt/move-wsl#23]
**Describe the bug**  
If the destination import directory is marked as "compressed", the import stage will fail due to `.vhdx` files not being compressable.

**Please add terminal output if possible:**  

```
jdsmi@YAMI ❯ ~\dev\scripts ❯
❯ .\Move-WSL.ps1
Getting distros...
Select distro to move:
1: docker-desktop
2: Arch
3: docker-desktop-data
1
Enter WSL target directory:
F:\docker\
Move docker-desktop to "F:\docker"? (Y|n): Y
Exporting VHDX to "F:\docker\docker-desktop.tar" ...
Unregistering WSL ...
Importing docker-desktop from F:\docker...
The requested operation could not be completed due to a virtual disk system limitation.  Virtual hard disk files must be uncompressed and unencrypted and must not be sparse.
Write-Error: Import failed. Distro not found. Export file at F:\docker\docker-desktop.tar
```

_Disabled compression on the destination folder._

```
⨯ jdsmi@YAMI ❯ ~\dev\scripts ❯
❯ wsl --import docker-desktop F:\docker F:\docker\docker-desktop.tar
jdsmi@YAMI ❯ ~\dev\scripts ❯
❯
```

**Script version:**
 - [ ] Bash
 - [x] Powershell
 - [ ] Batch

**Comment by sumodx** [comment]
Please upgrade severity, it wiped my WSL vhdx.

Repro:

1. Select compressed folder as target
2. Get Error as described above.
3. At this stage the WSL distro is unregistered and the source vhdx has been remove
4. Uncompress target folder (containing the tar)
5. Run BAT script again
6. Discover its deleted the tar file, and there is no source file (ref Step 3)

**Comment by pxlrbt** [comment]
> 5. Run BAT script again
> 6. Discover its deleted the tar file, and there is no source file (ref Step 3)

How did that tar file get removed? I'd expect the second run of the script to stop after exporting, since there is nothing to export anymore? 🤔 And in that case you should be able to manually import that .tar file.

Do you have any logs of the two runs?

**Comment by sumodx** [comment]
Sorry didnt see any logs :-(  I just reported my sequence of actions.

Perhaps whenever you pick up this issue you could add to your tests to see if you get a repro?

gl

**Comment by SayakMukhopadhyay** [comment]
I just tried reproducing this issue but could not. The distro got imported from a folder marked compressed just fine. Can one of you explain what you mean by compressed folder? As for me, I just went ahead and checked this option.
![image](https://user-images.githubusercontent.com/13748882/163775374-d5c21422-a1b7-459a-b1bd-3a7ed957ce1c.png)

**Comment by Kage-Yami** [comment]
> @SayakMukhopadhyay:
> I just tried reproducing this issue but could not. The distro got imported from a folder marked compressed just fine. Can one of you explain what you mean by compressed folder?

I have no idea if this is still an issue, as I don't use WSL at this time, but the issue is when the _destination_ is compressed, not the source folder from which your import `.tar` is in. 

From memory, the core of the problem is that if the destination folder is marked as compressed, then Windows attempts to compress the newly-created `.vhdx` file (the would-be post-move distro), which is not allowed.

**Comment by SayakMukhopadhyay** [comment]
> the core of the problem is that if the destination folder is marked as compressed, then Windows attempts to compress the newly-created .vhdx file

Yeah, I verified and that seems to be the case but it imported just fine in my case.
![image](https://user-images.githubusercontent.com/13748882/163777179-6144f9b9-de28-40e1-ae15-e352508ada0d.png)
 I have also ran the distro and it worked just fine. Do you have any further ideas?

**Comment by Kage-Yami** [comment]
> @SayakMukhopadhyay
> Do you have any further ideas?

Nothing off the top of my head, other than perhaps Microsoft has removed the limitation in Windows 11?

**Comment by SayakMukhopadhyay** [comment]
I found this issue https://github.com/microsoft/WSL/issues/4103 but weirdly I still can't reproduce this.
